Modernize modular scale

scottkellum opened the pull request:
Bring Modular Scale up to current Sass standards.

- Switch to @use from @import to more safely scope the library #161
- Use new math module
- Update responsive logic to use [Typetura](https://typetura.com), clamp(), and updated CSS Locks logic #148 #155 #168
- Change division method to the new standard #171

scottkellum committed:
Cleanup

scottkellum committed:
Starting the re-write

scottkellum committed:
Chipping away

scottkellum committed:
Updates

scottkellum committed:
Working on multi stranded scales

scottkellum committed:
Gitting the core function working

scottkellum committed:
More iteration

scottkellum committed:
Target feature done

scottkellum committed:
Working on various responsive solutions

scottkellum committed:
Run task for Nova and respond tweak

scottkellum committed:
Update Sass version number

scottkellum committed:
Prototype responsive logic

scottkellum committed:
Basic responsive logic in place

scottkellum committed:
Finished responsive logic

scottkellum committed:
Naming tweaks

scottkellum committed:
Tweaks and testing

scottkellum committed:
Missed a bunch of division errors. Fixed now

scottkellum committed:
Readme updates

scottkellum commented:
Please test this out here: https://github.com/modularscale/modularscale-sass/releases/tag/v4.0.0.rc1

scottkellum committed:
Version up

robsonsobral commented:
Hi, Scott!

Why not to move the `_modularscale.scss` to `modularscale/_index.scss`? The `@use` stays the same, but the files are kept all within a single folder.

I'm sorry for bother you twice in a day.

scottkellum commented:
@robsonsobral Not a bother! Happy to get feedback!

scottkellum committed:
Update configuration and loading

scottkellum committed:
Update version

scottkellum committed:
Document forwarding of settings functionality

robsonsobral committed:
Update `gitignore`

robsonsobral committed:
Install True Sass tester

robsonsobral committed:
Add test scripts

robsonsobral committed:
Start to add tests

robsonsobral committed:
Remove wrong parameter on test scripts

Why I thought I would need that?!

robsonsobral committed:
Prefix True

robsonsobral committed:
Rename `unitless()` to `strip-unit()` to...

...avoid conflicts with out reserved word.

https://sass-lang.com/documentation/modules/math#is-unitless

robsonsobral committed:
Test `strip-unit()`

robsonsobral committed:
Normalize IDE configurations

robsonsobral committed:
Improve `sort()` tests

robsonsobral committed:
Improve `strip-unit()` test

robsonsobral committed:
Test `to-num()` function

robsonsobral committed:
Reorganize tests to make proper use of `--watch`...

...and fix NPM test scripts

robsonsobral committed:
Test `target()` function

robsonsobral committed:
Add report to tests

robsonsobral committed:
Fix `/test-results` path

robsonsobral committed:
Add tests to `step()` function

robsonsobral committed:
Allow the use of target as "step"

robsonsobral committed:
Accept incompatible units on `step(12px)` using 16px as root size

robsonsobral committed:
Remove use of `@import`

robsonsobral committed:
Improve reversed steps tests

robsonsobral committed:
Test settings normalizer

robsonsobral committed:
Improve test script

robsonsobral committed:
Add multiple scales to `step()` test

robsonsobral committed:
Lint the code

robsonsobral committed:
Rename "vars" as "constants"

robsonsobral committed:
Kill a gremlin

https://marketplace.visualstudio.com/items?itemName=nhoizey.gremlins

robsonsobral committed:
Remove redundant code...

...because that's already the default settings

robsonsobral committed:
Fix multi bases `step()`

robsonsobral committed:
Normalize quotation marks

robsonsobral committed:
Call ratios by name...

...in a shorter way than the previous one

robsonsobral committed:
Allow target to be provided as list, using spaces

robsonsobral committed:
Prefix `type-of()`

robsonsobral committed:
Rename `target()` to make its purpose clearer

robsonsobral committed:
Move `step-by-target()` to its own file

robsonsobral committed:
Rename `settings()` to `normalize-settings()`

robsonsobral committed:
Allow adding custom named ratios...

..within settings map

robsonsobral committed:
Fix settings inheritance

robsonsobral committed:
Combine API into one file to make easier to handle settings

robsonsobral committed:
Fix settings overriding

robsonsobral committed:
Testing responsive mixins

scottkellum committed:
Merge pull request #174 from robsonsobral/4.x

robsonsobral commented:
@scottkellum ? Have you given it up?

scottkellum commented:
@robsonsobral Ha yeah I have no time for this for the next few months 😅🙃

robsonsobral commented:
I think we have a good thing here, @scottkellum . Let's give it some time?

scottkellum commented:
Just want to let you know I see this and am trying to figure out how to allocate time to wrap this up.

Lukortech commented:
I wanted to bump it a bit as I have some deprecation warnings while using this fantastic tool. 
If there's any way I could help out let me know @scottkellum

robsonsobral commented:
@Lukortech , I'm so sorry for letting you (and @scottkellum ) down.

I saw your message and planned to use a Brazilian holiday last week to take care of things, but I got sick.

Can you give some time to update the code?

scottkellum commented:
I feel that if I wade into this, I will only be a bottleneck. I’ve been drifting further and further away from this project. @robsonsobral and @Lukortech, feel free to merge PRs and push changes as you see fit. A code review might be wanted, so @Lukortech, maybe you can offer that when @robsonsobral is ready?

robsonsobral commented:
I've updated the syntax, @Lukortech ! Now, we're using the latest versions of SASS and True! Now, I need to review things, specially the related to @scottkellum 's Typetura, as I don't know it.

scottkellum commented:
Typetura stuff should probably be removed at this point.

Lukortech commented on readme.md:
```diff
@@ -8,76 +8,59 @@ To get started, you need to select a ratio and a base value. The base value is u
-* Terminal: `npm install modularscale-sass --save-dev`
-* SCSS: `@import 'modularscale'`
+- Terminal: `npm install modularscale-sass --save-dev`
+- SCSS: `@use '~modularscale-sass/stylesheets/modularscale' as ms;`
```
The long awaited change 🚀

scottkellum commented:
@Lukortech Thank you for your review. I added you as a maintainer if you wish to take a more active role.

@robsonsobral are we ready to go?

robsonsobral commented:
@scottkellum , I opened 2 issues that need fixing, and I'm working on them.

@Lukortech , sorry, but I can only work on this project on weekends.
